src/cmd/services/m3coordinator/x/tags_iterator.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```

Comment:
Hm, ideally we'd move to this into the aggregator repo.  But perhaps for now we should move this to `src/x/serialize` because it doesn't really live in the `src/cmd/services/m3coordinator/...` tree if it's going to become re-useable?

Reply:
Also actually, if we are going to be using the native `m3aggregator` binary for this flow, I don't think we need to touch this at all actually?  This was only a way to efficiently create IDs inside the `m3coordinator` itself, it was never supposed to be used in a full setup.

Reply:
I think moving src/x/serialize sounds better to me, cause this logic does not belong to m3aggregator, this decoder should be matched with the encoder used in collector. M3agg itself does not care about how the id is encoded/decoded from tags. 

I need this decoder for m3coordinator ingestion stuff because I'm getting ids from m3aggregator and need to convert that id into tags before writing into m3db. Unless that is not necessary? I saw the flush logic in downsampler is decoding ids into tags tho.

Reply:
Yes I guess what I mean though is that collector is going to encode these IDs and tags, not like how the m3coordinator was doing it.